library/memory_buffer_alloc.c

```diff
@@ -247,6 +249,8 @@ static void *buffer_alloc_calloc( size_t n, size_t size )
 
     if( n != 0 && len / n != size )
```

Comment:
In https://github.com/ARMmbed/mbedtls/issues/639 it was proposed that if the size of the requested allocation is `0` then the allocator should return `NULL`. The C99 standard states the following:
```
If the  size  of  the  space  requested  is  zero,  the  behavior  is  implementation-
defined:  either  a  null  pointer  is  returned,  or  the  behavior  is  as  if  the  size  were  some
nonzero value, except that the returned pointer shall not be used to access an object.
```
The problem is that some parts of the library assume that allocating 0 will return a non-NULL value on success. Making this change causes test failures, so the behaviour is not changed as it is compliant with the C standard.